lib/jasmine/jasmine.ts

```diff
@@ -1,32 +1,97 @@
 'use strict';
-// Patch jasmine's it and fit functions so that the `done` wrapCallback always resets the zone
-// to the jasmine zone, which should be the root zone. (angular/zone.js#91)
-if (!Zone) {
-  throw new Error('zone.js does not seem to be installed');
-}
-// When you have in async test (test with `done` argument) jasmine will
-// execute the next test synchronously in the done handler. This makes sense
-// for most tests, but now with zones. With zones running next test
-// synchronously means that the current zone does not get cleared. This
-// results in a chain of nested zones, which makes it hard to reason about
-// it. We override the `clearStack` method which forces jasmine to always
-// drain the stack before next test gets executed.
-(<any>jasmine).QueueRunner = (function (SuperQueueRunner) {
-  const originalZone = Zone.current;
-  // Subclass the `QueueRunner` and override the `clearStack` method.
+(() => {
+  // Patch jasmine's it and fit functions so that the `done` wrapCallback always resets the zone
+  // to the jasmine zone, which should be the root zone. (angular/zone.js#91)
+  if (!Zone) throw new Error("Missing: zone.js");
+  const SyncTestZoneSpec: {new (name: string): ZoneSpec} = Zone['SyncTestZoneSpec'];
+  if (!SyncTestZoneSpec) throw new Error("Missing: SyncTestZoneSpec");
+  const ProxyZoneSpec: {new (): ZoneSpec} = Zone['ProxyZoneSpec'];
+  if (!ProxyZoneSpec) throw new Error("Missing: ProxyZoneSpec");
```

Comment:
Under what circumstances will end users see the `"Missing: ProxyZoneSpec"` and `"Missing: SyncTestZoneSpec"` errors? They aren't particularly helpful presently.